Cognite.Simulator.Utils/Logger.cs

```diff
@@ -0,0 +1,148 @@
+        /// </summary>
+        /// <param name="config">Configuration object of <see cref="LoggerConfig"/> type</param>
+        /// <returns>A configured logger</returns>
+        public static Serilog.ILogger GetConfiguredLogger(LoggerConfig config)
+        {
+            var logConfig = LoggingUtils.GetConfiguration(config);
+            logConfig.WriteTo.Sink(sink);
+            logConfig.Enrich.With<UtcTimestampEnricher>();
+            logConfig.Enrich.FromLogContext();
+            return logConfig.CreateLogger();
+        }
+
+        /// <summary>
+        /// Flushes the stored logs.
+        /// </summary>
+        public static void FlushScopedRemoteApiLogs()
+        {
+
+            ((ScopedRemoteApiSink) sink).Flush();
+        }
+
+        /// <summary>
+        /// Create a default console logger and returns it.
+        /// </summary>
+        /// <returns>A <see cref="Microsoft.Extensions.Logging.ILogger"/> logger with default properties</returns>
+        public static Microsoft.Extensions.Logging.ILogger GetDefault() {
+            using (var loggerFactory = new LoggerFactory())
+            {
+                loggerFactory.AddSerilog(GetSerilogDefault(), true);
+                return loggerFactory.CreateLogger("default");
+            }
+        }
+
+    }
+   
+    /// <summary>
+    /// Extension utilities for logging
+    /// </summary>
+    public static class LoggingExtensions {
+
+        /// <summary>
+        /// Flushes the stored logs.
+        /// </summary>
+        public static void FlushScopedRemoteApiLogs(this Microsoft.Extensions.Logging.ILogger _)
+        {
+            SimulatorLoggingUtils.FlushScopedRemoteApiLogs();
+        }
+
+        /// <summary>
+        /// Adds a configured Serilog logger as singleton of the <see cref="Microsoft.Extensions.Logging.ILogger"/> and
+        /// <see cref="Serilog.ILogger"/> types to the <paramref name="services"/> collection.
+        /// A configuration object of type <see cref="LoggerConfig"/> is required, and should have been added to the
+        /// collection as well.
+        /// </summary>
+        /// <param name="services">The service collection</param>
+        /// <param name="buildLogger">Method to build the logger.
+        /// <param name="alternativeLogger">True to allow alternative loggers, i.e. allow config.Console and config.File to be null</param>
+        /// This defaults to <see cref="SimulatorLoggingUtils.GetConfiguredLogger(LoggerConfig)"/>,
+        /// which creates logging configuration for file and console using
+        /// <see cref="LoggingUtils.GetConfiguration(LoggerConfig)"/></param>
+        public static void AddLogger(this IServiceCollection services, Func<LoggerConfig, Serilog.ILogger> buildLogger = null, bool alternativeLogger = false)
+        {
+            var serviceProvider = services.BuildServiceProvider();
+            var cogniteDestination = serviceProvider.GetService<CogniteDestination>();
```

Comment:
Ideally, we should not build a service provider here, and use `p` to get the destination within the `AddSingleton<ILogger>` block.